File: Week 6/sentimental-cash/cash.py
def cal_q(cents):
    return cents // 25


def cal_d(cents):
    return cents // 10


def cal_n(cents):
    return cents // 5


def cal_p(cents):
    return cents


def main():
    while True:
        try:
            dollars = float(input("Change owed: "))
            if dollars >= 0:
                break
        except ValueError:
            pass

    cents = round(dollars * 100)

    # Calculate coins
    q = cal_q(cents)
    cents -= q * 25

    d = cal_d(cents)
    cents -= d * 10

    n = cal_n(cents)
    cents -= n * 5

    p = cal_p(cents)
    cents -= p * 1
    # Total coins
    cash = q + d + n + p
    print(cash)


if __name__ == "__main__":
    main()

File: Week 1/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int cal_q(int cents);
int cal_d(int cents);
int cal_n(int cents);
int cal_p(int cents);

int main(void)
{
    int cents;
    do
    {
        cents = get_int("Changed owed: ");
    }
    while (cents < 0);

    int q = cal_q(cents);
    cents = cents - (q * 25);

    int d = cal_d(cents);
    cents = cents - (d * 10);

    int n = cal_n(cents);
    cents = cents - (n * 5);

    int p = cal_p(cents);
    cents = cents - (p * 1);

    int cash = q + d + n + p;
    printf("%d\n", cash);
}
int cal_q(int cents)
{
    int q = 0;

    while (cents >= 25)
    {
        q++;
        cents = cents - 25;
    }
    return q;
}
int cal_d(int cents)
{
    int d = 0;
    while (cents >= 10)
    {
        d++;
        cents = cents - 10;
    }
    return d;
}
int cal_n(int cents)
{
    int n = 0;
    while (cents >= 5)
    {
        n++;
        cents = cents - 5;
    }
    return n;
}
int cal_p(int cents)
{
    int p = 0;
    while (cents >= 1)
    {
        p++;
        cents = cents - 1;
    }
    return p;
}
